validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -464,6 +476,31 @@ def remove_temp_endpoint(self, endpoint):
         if endpoint in self._temp_endpoints:
             del self._temp_endpoints[endpoint]
 
+    def _check_temp_endpoints(self):
+        for endpoint in self._temp_endpoints:
+            endpoint_info = self._temp_endpoints[endpoint]
+            if (time.time() - endpoint_info.time) > \
+                    endpoint_info.retry_threshold:
+                LOGGER.debug("Endpoint has not completed authorization in %s "
+                             "seconds. Retry endpoint: %s ",
+                             endpoint_info.retry_threshold,
+                             endpoint)
+                try:
+                    conn_id = \
+                        self._network.get_connection_id_by_endpoint(endpoint)
+                    self._network.remove_connection(conn_id)
+                except KeyError:
+                    pass
+
+                self._network.add_outbound_connection(endpoint)
+                self._temp_endpoints[endpoint] = EndpointInfo(
+                    endpoint_info.status,
+                    time.time(),
+                    endpoint_info.retry_threshold * 2 if
```

Comment:
This whole block can be `min(endpoint_info.retry_threshold * 2, MAXIMUM_RETRY_FREQUENCY)`